packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -179,36 +179,18 @@ export class AbstractQueryGeneratorTypeScript {
     }
 
     if (options) {
-      const CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS = new Set<keyof CreateSchemaQueryOptions>();
-      if (this.dialect.supports.createSchema.authorization) {
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS.add('authorization');
-      }
-
-      if (this.dialect.supports.createSchema.charset) {
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS.add('charset');
-      }
-
-      if (this.dialect.supports.createSchema.collate) {
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS.add('collate');
-      }
-
-      if (this.dialect.supports.createSchema.comment) {
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS.add('comment');
-      }
-
-      if (this.dialect.supports.createSchema.ifNotExists) {
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS.add('ifNotExists');
-      }
-
-      if (this.dialect.supports.createSchema.replace) {
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS.add('replace');
-      }
-
       rejectInvalidOptions(
         'createSchemaQuery',
-        this.dialect.name,
+        this.dialect,
         CREATE_SCHEMA_QUERY_SUPPORTABLE_OPTIONS,
-        CREATE_SCHEMA_QUERY_SUPPORTED_OPTIONS,
+        {
+          authorization: this.dialect.supports.createSchema.authorization,
+          charset: this.dialect.supports.createSchema.charset,
+          collate: this.dialect.supports.createSchema.collate,
+          comment: this.dialect.supports.createSchema.comment,
+          ifNotExists: this.dialect.supports.createSchema.ifNotExists,
+          replace: this.dialect.supports.createSchema.replace,
+        },
```

Comment:
In cases like these, why don't we pass `this.dialect.supports.createSchema` directly?

Reply:
Oh yeah, I did not realize it was a one to one match

Reply:
Done!